kv/txn.go

```diff
@@ -54,7 +60,7 @@ func RunInNewTxn(store Storage, retryable bool, f func(txn Transaction) error) e
 			return errors.Trace(err)
 		}
 
-		err = txn.Commit(context.Background())
+		err = txn.Commit(context.WithValue(context.Background(), Retryable, retryable))
```

Comment:
I really dislike passing those value through context, even `txn.SetOption(retryable)` is better than this.
If you insist, I'll refactor the code here after it's merged.

Reply:
I've tried your suggestion and it's a little complex. I think to refactor the code in another PR is 
 ok for me.